cmd/tsidp/tsidp.go

```diff
@@ -611,6 +758,124 @@ func (s *idpServer) serveOpenIDConfig(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// funnelClient represents an OIDC client/relying party that is accessing the
+// IDP over Funnel.
+type funnelClient struct {
+	ID     string `json:"client_id"`
+	Secret string `json:"client_secret,omitempty"`
+	Name   string `json:"name,omitempty"`
+}
+
+// /clients is a privileged endpoint that allows the visitor to create new
+// Funnel-capable OIDC clients, so it is only accessible over the tailnet.
+func (s *idpServer) serveClients(w http.ResponseWriter, r *http.Request) {
+	if isFunnelRequest(r) {
```

Comment:
why is this not doing the whois check itself?

Reply:
Can we rely on the whois check regardless of if the call is coming over funnel or on the tailnet directly?

Reply:
the whois check for a call coming over funnel just results in `funnel-ingress-node`